### ISSUE TITLE
Edge is not debugging SPfx web part in hosted workbench

### ISSUE THREAD

**Issue by ARaheem-Work** [SharePoint/sp-dev-docs#7465]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version: Version 95.0.1020.40 (Official build) (64-bit)
- SPFx version: 1.13.0
- Node.js version: 14.15.0
- VSCode: 64-bit, Version 1.61.2
- Windows OS: Windows 10 Enterprise Evaluation. Build 19041.vb_release.191206-1406

### Describe the bug / error

I have followed all the steps as per [debug in SPFx](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/debug-in-vscode), but could not attach edge browser to debugger session. I attempted with both profiles, in particular, Hosted workbench Edge. Launch.json file is attached as .txt file.

SPFx web part is fairly simple, code compiles successfully and I'm able to see web part in the list of web parts, add it to the hosted work bench and render it. However not able to debug it using Edge browser. I tried chrome as well, then dropped idea because chrome extension has been deprecated.

Is there anything I am missing ?
Thanks for all the help.

--------------
A few observations:

1.  I am able to see and load web part in hosted workbench when I run command: gulp server
2. I'm not able to see web part web part list when I click F5 in debug session. I get a message on hosted workbench page. Attachment: workbench-message.png.

-----------------
[workbench-message](https://user-images.githubusercontent.com/92548775/139742841-d4b82a13-7924-4a75-9c71-d3bd53917138.png)
[Launch-backup.txt](https://github.com/SharePoint/sp-dev-docs/files/7456174/Launch-backup.txt)
)

### Steps to reproduce

1.Add a breakpoint in Render method in web part code file.
2. gulp serve --nobrowser
3. Select debug profile: Hosted workbench Edge and click F5.

### Expected behavior

Edge browser should attach to VS code debug session.

**Comment by patmill** [comment]
So, just tried this out, and had no problems.  I _did_ change the .vscode/launch.json file to have 

      "type":"msedge",

and updated the url to point to my tenant.  Hit F5 and added the webpart to the canvas.  The render breakpoint was hit.  Note that I used "msedge", not "edge"

**Comment by patmill** [comment]
I've updated the docs to use msedge

**Comment by ARaheem-Work** [comment]
Hi @patmill It looks like this time I am stuck on a step before debugging. So this time I am not able to see web part in hosted workbench (I tried both chrome and edge browsers to verify web part, but no luck). I used following commands. Please help. Thanks.
gulp serve
gulp serve --nobrowse

SPFx version: 1.13.0
![web-part-not-loading-hosted-workbench](https://user-images.githubusercontent.com/92548775/140177337-f77f95b0-e360-4d34-a572-aa24ae60b9c7.PNG)
![gulp-serve-command-output](https://user-images.githubusercontent.com/92548775/140177361-e8ac543c-4d76-46c9-9504-247485bec9b6.PNG)
r

**Comment by patmill** [comment]
You should only be running serve once, not twice.

But once you do that, and you wait until the build is finished and _then_ load the hosted workbench, same issue?  Normally you should see a request in your console when the hosted workbench tries to see what parts are being locally hosted, and I'm not seeing that.  Makes me think that the

**Comment by ARaheem-Work** [comment]
@patmill 
Yes, sure. What I meant, I ran both at different times, but didn't change behavior.

Ok, so I ran gulp serve, it opened hosted workbench automatically based on url set in \config\serve.json file. Got same web part not loading warning dialog box. Waited for build process to finish (screenshot attached). Loaded hosted workbench again, same warning dialog.

I do have multiple browser profiles in Microsoft Edge browser. Is that going to cause problem with SPFx ?

Thanks
![gulp-serve-command-output-2](https://user-images.githubusercontent.com/92548775/140180676-f97a2291-5d3f-4245-893a-ee898cff05f3.PNG)

**Comment by patmill** [comment]
So, again.  The problem is that you are getting that warning message, which means that you aren't actually serving the code from your project.

After gulp serve is done building / webpack / etc. and you run your hosted workbench, you should see a network call made to https://localhost:4321/temp/manifests.js

What is the response on it?

in your console, you should have a line like this after you hit f5:
![image](https://user-images.githubusercontent.com/3767881/140193149-bc824a89-6c60-4d2a-a4a6-88a3a6404588.png)

**Comment by ARaheem-Work** [comment]
Hi @patmill 

Thanks for all the help. I was able to load web part in hosted workbench, start debugging session and hit breakpoint as well.
I have added a screenshot below for future reference, showing mapping of workbench to network trace.

Appreciate all the help. 

This request can be closed now.

Thanks
![gulp-serve-command-output-4](https://user-images.githubusercontent.com/92548775/140235376-6fc8a194-7801-4f68-a0a4-234c8fe36c58.png)
